enterprise/trackedusers/users_reporter.go

```diff
@@ -143,6 +145,23 @@ func (u *UniqueUsersReporter) GenerateReportsFromJobs(jobs []*jobsdb.JobT, sourc
 			combinedUserIDAnonymousID := combineUserIDAnonymousID(userID, anonymousID)
 			workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID] = u.recordIdentifier(workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID], combinedUserIDAnonymousID, idTypeIdentifiedAnonymousID)
 		}
+
+		// for alias event we will be adding previousId to identifiedAnonymousID hll,
+		// so for calculating unique users we do not double count the user
+		// e.g. we receive events
+		// {type:track, anonymousID: anon1}
+		// {type:track, userID: user1}
+		// {type:track, userID: user2}
+		// {type:identify, userID: user1, anonymousID: anon1}
+		// {type:alias, previousId: user2, userID: user1}
+		// userHLL: {user1, user2}, anonHLL: {anon1}, identifiedAnonHLL: {user1-anon1, user2}
+		// cardinality: len(userHLL)+len(anonHLL)-len(identifiedAnonHLL): 2+1-2 = 1
+		if eventType == eventTypeAlias {
+			previousID := gjson.GetBytes(job.EventPayload, "batch.0.previousId").String()
```

Comment:
Why are we only looking at the first entry in `batch` and not all entries ?

Reply:
Each batch will have only have 1 event. Even if on gateway we receive a batch request we break that down and we create a new entry for each event in a batch. and each entry will be like `{"batch":[{event...}]}`